### PR TITLE
Always use CLDR locale on ES v9

### PR DESCRIPTION
**Pull request by thecoop**
Always use the CLDR locales on v9+

### CODE REVIEW

File: distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/SystemJvmOptions.java [hunk 1]
```diff
@@ -73,16 +72,6 @@ static List<String> systemJvmOptions(Settings nodeSettings, final Map<String, St
...
     }
 
-    @UpdateForV9    // only use CLDR in v9+
-    private static String getLocaleProviders() {
-        /*
-         * Specify SPI to load IsoCalendarDataProvider (see #48209), specifying the first day of week as Monday.
-         * When on pre-23, use COMPAT instead to maintain existing date formats as much as we can.
-         * When on JDK 23+, use the default CLDR locale database, as COMPAT was removed in JDK 23.
-         */
-        return Runtime.version().feature() >= 23 ? "SPI,CLDR" : "SPI,COMPAT";
```
Comment (rjernst) [inline distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/SystemJvmOptions.java:83]: Should the change (which would be backported to 8.x) to use just CLDR for jdk 23 be made first?

Comment (thecoop) [reply distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/SystemJvmOptions.java:83]: See https://github.com/elastic/elasticsearch/pull/113182

**Comment by thecoop** [comment]
@elasticmachine update branch

**Comment by thecoop** [comment]
@elasticmachine rerun elasticsearch-ci/part-4

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**